lib/elixir/src/elixir_parser.yrl

```diff
@@ -939,11 +939,11 @@ build_access(Expr, {List, Meta}) ->
 
 %% Interpolation aware
 
-build_sigil({sigil, Location, Sigil, Parts, Modifiers, Indentation, Delimiter}) ->
+build_sigil({sigil, Location, _SigilName, Atom, Parts, Modifiers, Indentation, Delimiter}) ->
```

Comment:
Do we need the SigilName now? Perhaps we can only pass the Atom and build the sigil name from the atom where necessary?

Reply:
The only place it's used is [here](https://github.com/elixir-lang/elixir/pull/12706/files#diff-7b8b9e97cda7c98a730da8dabeeab44342fa87bc4986dbd54fd91b315ff94b38R251), but it felt weird trying to reverse engineer the original chars especially since in case of a static encoder `Atom` can actually be anything and so the information would be lost.
It might be a reasonable trade-off to have a degraded error message if we fail to have a proper atom starting with `sigil_`: see https://github.com/elixir-lang/elixir/pull/12706/commits/d36ec6046323f80dad514dd540203b521799b96a. WDYT?